From 09308141f9cfd34e0b12024faea78e6f1a6dcb0d Mon Sep 17 00:00:00 2001
From: Ronan Abhamon <ronan.abhamon@vates.fr>
Date: Mon, 21 Oct 2024 18:44:42 +0200
Subject: [PATCH] Count correctly IQN sessions during ISCSISR attach

Before this change, IQNs were concatenated into a single string when `multiSession` was used.

Signed-off-by: Ronan Abhamon <ronan.abhamon@vates.fr>
---
 drivers/BaseISCSI.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/BaseISCSI.py b/drivers/BaseISCSI.py
index 750b40b0..e6092514 100755
--- a/drivers/BaseISCSI.py
+++ b/drivers/BaseISCSI.py
@@ -391,10 +391,9 @@ class BaseISCSISR(SR.SR):
         util._incr_iscsiSR_refcount(self.targetIQN, sr_uuid)
         IQNs = []
         if "multiSession" in self.dconf:
-            IQNs = ""
             for iqn in self.dconf['multiSession'].split("|"):
                 if len(iqn):
-                    IQNs += iqn.split(',')[2]
+                    IQNs.append(iqn.split(',')[2])
         else:
             IQNs.append(self.targetIQN)
 
